src/L2VoteAggregator.sol

```diff
@@ -67,12 +82,14 @@ abstract contract L2VoteAggregator {
   mapping(uint256 proposalId => ProposalVote) public proposalVotes;
 
   /// @dev Emitted when a vote is cast on L2.
-  event VoteCast(address indexed voter, uint256 proposalId, VoteType support, uint256 weight);
+  event VoteCast(
+    address indexed voter, uint256 proposalId, VoteType support, uint256 weight, string reason
+  );
 
   /// @param _votingToken The token used to vote on proposals.
   /// @param _governorMetadata The `GovernorMetadata` contract that provides proposal information.
   /// @param _l1BlockAddress The address of the L1Block contract.
-  constructor(address _votingToken, address _governorMetadata, address _l1BlockAddress) {
+  constructor(address _votingToken, address _governorMetadata, address _l1BlockAddress)  EIP712("L2VoteAggregator", "1") {
```

Comment:
It is fine if we hard code this? I was thinking we could use the protocol name. So maybe this should even be `L2FlexibleVoting`.

https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator